src/sentry/static/sentry/app/utils/integrationUtil.tsx

```diff
@@ -233,3 +240,30 @@ export const getCategories = (features: IntegrationFeature[]): string[] => {
 
   return [...new Set(transform)];
 };
+
+export const getCategoriesForIntegration = (
+  integration: AppOrProviderOrPlugin
+): string[] => {
+  if (isSentryApp(integration)) {
+    const features = ['internal', 'unpublished'].includes(integration.status)
+      ? [{featureGate: integration.status, description: ''}]
```

Comment:
@NisanthanNanthakumar I am not a fan of creating this fake integration feature like this. Why can't we just do:
```
return ['internal', 'unpublished'].includes(integration.status) ? integration.status : getCategories(integration.featureData)
```